portal/app/api/v1/endpoint/farm_device.py

```diff
@@ -20,4 +27,23 @@ def get_queryset(self, **kwargs):
         return queryset
 
 
+class FarmDataViewSet(viewsets.ModelViewSet):
+    serializer_class = FarmDataSerializer
+    authentication_classes = (CsrfExemptAuth.CsrfExemptSessionAuthentication, TokenAuthentication)
+
+    def get_queryset(self, **kwargs):
+        device_uid = self.request.query_params.get('device_uid', None)
+        start_date = self.request.query_params.get('start_date', None)
+        end_date = self.request.query_params.get('end_date', None)
+        if device_uid is None or start_date is None or end_date is None:
+            raise InvalidParamException
+            return
+        queryset = FarmData.objects.filter(farm_device__device_uid=device_uid, timestamp__range=[start_date, end_date])
+        return queryset
+
 
+class InvalidParamException(APIException):
+    status_code = 404
```

Comment:
This should really be a 400 instead of a 404. Lookup the diff and see what you think.

Reply:
I meant 400 - typo. Just fixed